autoPyTorch/datasets/base_dataset.py

```
@@ -68,10 +67,12 @@ def __init__(
         self.shuffle = shuffle
         self.resampling_strategy = resampling_strategy
         self.resampling_strategy_args = resampling_strategy_args
-        self.task_type: Optional[int] = None
+        self.task_type: Optional[str] = None
         self.issparse: bool = issparse(self.train_tensors[0])
-        self.output_type: int = STRING_TO_OUTPUT_TYPES[type_of_target(self.train_tensors[1])]
-
+        self.num_classes = None
+        if len(train_tensors) == 2 and train_tensors[1] is not None:
```

What happens with `num_classes` and `output_type` if this is not true?

---

That would be the case when we're doing unsupervised learning. currently, I am not taking care of that as I focused on classification and regression. What would you suggest to do? Should I create output constants for unsupervised learning as well? Or for now, we can also raise an error saying that the task is currently not supported.

---

Then let's just create an `else: raise NotImplementedError("")` explaining that